src/cmap/establish/mod.rs

```diff
@@ -59,14 +65,29 @@ impl ConnectionEstablisher {
             None
         };
 
+        let connect_timeout = if let Some(timeout) = options.connect_timeout {
```

Comment:
Nit: I think this might be easier to read as a match against `Some(0)`, `Some(n)`, and `None`.

Reply:
Good idea, done. I had to do it a little bit different since we're matching on a `Duration`, but I still think it's more readable than before.